api/admin/controller.py

```diff
@@ -203,13 +203,17 @@ def authenticated_admin_from_request(self, email=None, type=None):
     def authenticated_admin(self, admin_details):
         """Creates or updates an admin with the given details"""
 
-        admin, ignore = get_one_or_create(
+        admin, is_new = get_one_or_create(
             self._db, Admin, email=admin_details['email']
         )
         admin.update_credentials(
             self._db,
             credential=admin_details.get('credentials'),
         )
+        if is_new and admin_details.get("roles"):
+            for role in admin_details.get("roles"):
+                library = Library.lookup(self._db, role.get("library"))
+                admin.add_role(role.get("role"), library)
```

Comment:
Either this code or `add_role` should verify that the role being added is one of the actually existing roles.

Reply:
What should it do in that case? Log a warning and ignore the role?

Reply:
Yeah, that seems fine. It won't happen in real life so no need to make a ProblemDetail out of it.